MH4U Dex/CoreDataController.m

```diff
@@ -472,29 +484,70 @@ - (Quest *)questFromDictionary:(NSDictionary *)questDict withContext:(NSManagedO
     
     NSPredicate *monsterPredicate = [NSPredicate predicateWithFormat:@"%K == %@", MonsterAttributes.name, questDict[MHDQuestFirstMonsterKey]];
     // The same is the case with the following monster-fetching calls
-    quest.firstMonster = (Monster *)[self uniqueEntityWithEntityName:[Monster entityName]
-                                                       withPredicate:monsterPredicate
-                                                         withContext:context];
-    if (quest.firstMonster) {
-        //Continue on to the second, and so on.  By nesting this way, we can avoid unnecessary fetches
+    Monster *monster = (Monster *)[self uniqueEntityWithEntityName:[Monster entityName]
+                                                     withPredicate:monsterPredicate
+                                                       withContext:context];
+    if (monster) {
+        [quest.monsterSet addObject:monster];
         monsterPredicate = [NSPredicate predicateWithFormat:@"%K == %@", MonsterAttributes.name, questDict[MHDQuestSecondMonsterKey]];
-        quest.secondMonster = (Monster *)[self uniqueEntityWithEntityName:[Monster entityName]
-                                                            withPredicate:monsterPredicate
-                                                              withContext:context];
-        if (quest.secondMonster) {
+        monster = (Monster *)[self uniqueEntityWithEntityName:[Monster entityName]
+                                                withPredicate:monsterPredicate
+                                                  withContext:context];
```

Comment:
I wouldn't re-use the `monster` variable here when loading a different object, just to make sure you don't accidentally use the wrong one later on.